transports/vm/src/main/java/org/mule/transport/vm/VMMessageRequester.java

```diff
@@ -57,31 +56,31 @@ protected MuleMessage doRequest(long timeout) throws Exception
             }
             else
             {
-                MuleEvent event = null;
+                MuleMessage message = null;
                 if (logger.isDebugEnabled())
                 {
                     logger.debug("Waiting for a message on " + endpoint.getEndpointURI().getAddress());
                 }
                 try
                 {
-                    event = (MuleEvent) queue.poll(timeout);
+                    message = (MuleMessage) queue.poll(timeout);
```

Comment:
This poll assumes no events are left in a persistent queue when upgrading to this version.